### DISCUSSION TITLE
Removing Bouncy Castle dependency

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-milo/milo#797] [General]
Hello.  Is it possible to remove the dependency on Bouncy Castle in Milo and just use JDK8?  If so, is there any interest in doing it?

Thanks

**Comment by kevinherron** [comment]
You can try manually excluding the transitive dependency on BC in your build and then not using any of its functionality but I'm not interested in trying to remove it right now. 

Once upon a time it was actually only a soft dependency but I gave up on that because some of the security profiles don't work with the JCA provider and the certificate generation and validation stuff isn't possible using only Java APIs.

**Reply by ghost** [reply]
OK.  If I remove it, can it be integrated into the master branch so it becomes part of the product?  Either replacing BC or as a standard option (e.g. a run-time or build-time switch which lets you choose one or the other)?

**Reply by kevinherron** [reply]
No, the next major version I'm planning to release after 0.6.0 this week is 1.0 and there's no way you can remove BC without breaking APIs and creating new modules. If you need a release that doesn't depend on it you'll have to fork and keep it private.

Right now `stack-core` depends on it for certificate generation, certificate validation, and for the `CertificateUtil` class. You will have a difficult time replacing some of that functionality without using BC.

**Reply by ghost** [reply]
OK, thanks.

**Reply by kevinherron** [reply]
I will however keep this in mind as a stretch goal for version 2.0 which will break APIs all over again (not as far out as you think, going to start working on it immediately after 1.0 or possibly earlier).

What is your use case for removing the BC dependency?
